package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -10,6 +10,43 @@ local uci = require('simple-uci').cursor()
 -- Initial
 if not sysconfig.gluon_version then
 	uci:delete_all('wireless', 'wifi-iface')
+
+	-- First count all radios with a fixed frequency band.
+	-- This is needed to distribute devices which have radios
+	-- capable of operating in the 2.4 GHz and 5 GHz band need
+	-- to be distributed evenly.
+	local radio_band_count = {band24=0, band5=0}
+	util.foreach_radio(uci, function(radio, index, config)
+		local hwmodes = iwinfo.nl80211.hwmodelist(util.find_phy(radio))
+		if (hwmodes.a or hwmodes.ac) and hwmodes.g then
+			-- Dualband - do nothing in this step
+		elseif hwmodes.g then
+			-- 2.4 GHz
+			radio_band_count["band24"] = radio_band_count["band24"] + 1
```

Comment:
Hmm, we could use this code and clean up legacy hwmode values '11ng' and '11na' (that we have to handle in multiple places otherwise). How about using the strings '11n' and '11a' as map keys in *radio_band_count* - the increment + uci:set could be refactored into a local function that way, where only radio_name and one of these two strings need to be passed.

Reply:
Hmm, or not. At least when guarded by `if not sysconfig.gluon_version then`, this will obviously not help with upgrading legacy configurations...